### ISSUE TITLE
Design pagination UI and Technical spike

### ISSUE THREAD

**Issue by astroash** [TwinePlatform/DataPower#359]
- [x] Tech spike on paginating with postgres
- [x] Look into how table components are rendering
- [x] Create design options

**Comment by astroash** [comment]
@eliascodes @RogeredBacon 

## Suggested design:
<img width="773" alt="screen shot 2018-04-11 at 11 06 58 am" src="https://user-images.githubusercontent.com/14337958/38604287-873855b4-3d78-11e8-8577-b84a0f640e95.png">

OR in purple

<img width="771" alt="screen shot 2018-04-11 at 11 06 24 am" src="https://user-images.githubusercontent.com/14337958/38604291-891fd000-3d78-11e8-8cd3-d25f1371b489.png">

Let me know what you think. If you suggest another pagination tab layout please give picture references. There's a good examples list here : http://ui-patterns.com/patterns/Pagination/examples

**Comment by eliascodes** [comment]
IMO the orange one, but the purple one might be better for people who need higher contrast. Reckon we should ask Sonja?

**Comment by astroash** [comment]
Yeah, background colour is a minimal change so we can do that anytime. I was more looking for layout functionality feedback as this will be more of a pain to change later.

Moved it around a bit based feedback from Ghada:

<img width="773" alt="screen shot 2018-04-11 at 12 39 41 pm" src="https://user-images.githubusercontent.com/14337958/38609228-38d38c00-3d86-11e8-9b28-53adcfcfd40b.png">

**Comment by eliascodes** [comment]
Also it should fill the entire width of the table, with numbers evenly spaced. The "next" and "prev" buttons should be significantly bigger than the numbers.

**Comment by eliascodes** [comment]
@astroash what were your thoughts on how filtering impacts all this? From what I can see, we have three options:

1. Load all data, paginate and filter client-side, do not fetch more data
2. Load first page of data, fetch more data on page change and filter change
3. Load several pages of data, paginate and filter client-side, fetch more data when exhausted

In terms of server/DB load, I suspect option 1 is most efficient, and technically the simplest.

Thoughts?

**Comment by astroash** [comment]
@eliascodes I was leaning towards doing all pagination/filtering/sorting on server through db queries (option 2). This will increase server/db load however IMO is the most scalable option as it will allow us to handle 1000's of lines of data without effecting load time. 

Option 1 is only more efficient if we are working with small data sets. As our data grows the initial load time significantly increases from super inefficient js data handling, as well as requesting/sending over more data from the db. This option also relies on the processing power of _potentially dated_ client side devices. Arguably we could use this option for visitors as this is likely to be less data, however it would be my preference to use the same method for visitors and visits.

Option 3 I see as an optimisation measure you would take if option 2 proves problematic in production as it increases the complexity of how pagination is implemented.

TLDR
**SQL is made for handling data, so I'm game for letting it do it's thing.**

**Comment by eliascodes** [comment]
Spike is done.
